packages/shared/src/components/FeedItemComponent.tsx

```diff
@@ -163,8 +153,28 @@ export default function FeedItemComponent({
             ...item.post,
           }}
           data-testid="postItem"
-          onUpvoteClick={(post) => onUpvote(post, index, row, column)}
-          onDownvoteClick={(post) => onDownvote(post, index, row, column)}
+          onUpvoteClick={(post) => {
+            onUpvote({
+              post,
+              origin: Origin.Feed,
+              opts: {
+                columns,
+                column,
+                row,
+              },
+            });
+          }}
```

Comment:
Why is this one using the `useVotePost` if it's feedItemComponent.
Could we not use the `useFeedVotePost` here?

Reply:
Yes, but you would have to prop drill much more. Using base one allows us to just use without prop requirements and then mutation subscription catches it and invalidates feed cache from there.

I get the point it might be hard to grasp with subscriptions and basically event embitter but I think it works the best with our current cache setup.

For some kind of cache refactor I would advice keeping each post <-> user state in separate queryKey which would then allow invalidation without having access to whole feed or post index in feed array. But that is for some refactor down the line.

Reply:
I see :)